### PR TITLE
Parallelize unit test

### LINKED ISSUE #5565: Test suite are sensitive to run order
When trying to apply parallelization with `pytest-xdist` to the test suite (in `holoviews/pytest`), I noticed that it is very dependent on the run order.  

If the order is randomized with something like `pytest-random-order` and the following command: `pytest holoviews/tests --random-order-bucket=class`, the test suite will fail. 

Looking at the code a culprit for this could be having `Store.renderers['bokeh']` out in the open and not inside the `setUp` of the class.

### PR DESCRIPTION
**Pull request by hoxbro**
Fixes #5565

This time it has been in a much better state than when I looked over a year ago. I think this relates to work done in #5954 and #6200, where I made it possible to run tests with required dependencies + pytest. 

The main change in this PR is to reset `hv.Store` before each run.

### CODE REVIEW

File: holoviews/tests/plotting/matplotlib/test_renderer.py [hunk 1]
```diff
@@ -150,6 +150,7 @@ def test_render_holomap_individual_widget_position(self):
         self.assertEqual(obj.widget_location, 'top')
         self.assertEqual(obj.widget_type, 'individual')
 
+    @pytest.mark.filterwarnings('ignore:Attempted to send message over Jupyter Comm:UserWarning')
```
Comment (hoxbro) [inline holoviews/tests/plotting/matplotlib/test_renderer.py:153]: This is already done for the Bokeh backend

**Comment by hoxbro** [comment]
Locally, I have run the following script which uses `pytest-random-order` on https://github.com/holoviz/holoviews/pull/6234/commits/f0541c6cd2cbae2c65166003f1c2b12bdda5d4f9

``` bash
mkdir -p logs
for i in {1..100}; do
    echo "Run $i"
    pytest holoviews/tests -n logical --dist loadgroup --random-order --color=no > logs/log$i.txt
done
```

This resulted in the following:
``` console
❯ rg FAILED logs | sort
logs/log12.txt:FAILED holoviews/tests/ipython/test_magics.py::TestOptsMagic::test_cell_opts_plot_dynamic
logs/log28.txt:FAILED holoviews/tests/plotting/bokeh/test_callbacks.py::TestCallbacks::test_callback_cleanup
logs/log56.txt:FAILED holoviews/tests/ipython/test_magics.py::TestOptsMagic::test_cell_opts_plot_dynamic
logs/log57.txt:FAILED holoviews/tests/plotting/bokeh/test_callbacks.py::TestCallbacks::test_callback_cleanup
logs/log61.txt:FAILED holoviews/tests/plotting/bokeh/test_callbacks.py::TestCallbacks::test_callback_cleanup
logs/log72.txt:FAILED holoviews/tests/plotting/bokeh/test_server.py::TestBokehServer::test_server_dynamicmap_with_stream
logs/log78.txt:FAILED holoviews/tests/plotting/bokeh/test_server.py::TestBokehServer::test_server_dynamicmap_with_dims
logs/log81.txt:FAILED holoviews/tests/plotting/bokeh/test_server.py::TestBokehServer::test_server_dynamicmap_with_dims
logs/log94.txt:FAILED holoviews/tests/ipython/test_magics.py::TestOptsMagic::test_cell_opts_style_dynamic


❯ rg FAILED logs --json | jq '[select(.type == "match")][].data.path.text' | xargs rg seed | grep seed | sort
logs/log12.txt:Using --random-order-seed=94455
logs/log28.txt:Using --random-order-seed=484446
logs/log56.txt:Using --random-order-seed=735748
logs/log57.txt:Using --random-order-seed=876950
logs/log61.txt:Using --random-order-seed=72331
logs/log72.txt:Using --random-order-seed=438329
logs/log78.txt:Using --random-order-seed=47376
logs/log81.txt:Using --random-order-seed=747979
logs/log94.txt:Using --random-order-seed=479472
```

Running with those seeds again, the test did not fail, so I have marked them as flaky.

**Comment by hoxbro** [comment]
Isolated test failures (from https://github.com/holoviz/holoviews/issues/5565) 

| Status | Command | 
| --- | --- |
| :heavy_check_mark: | `pytest holoviews/tests -k InspectorTests`   
|  | `pytest holoviews/tests/core/data/test_imageinterface.py::RGBElement_ImageInterfaceTests::test_reduce_to_single_values` | 
| :heavy_check_mark: | `holoviews/tests/core/test_options.py::TestCrossBackendOptionSpecification::test_mpl_bokeh_mpl_via_builders_opts_method` |

**Review by philippjfr — APPROVED** [review]
Really nice to see!

**Pull request merged**